cmd/controller-gen/main.go

```diff
@@ -24,6 +24,7 @@ import (
 
 	"github.com/spf13/cobra"
 
+	"sigs.k8s.io/controller-tools/pkg/applyconfigurations"
```

Comment:
@Jefftree,

The only thing that needs to be added is tests. So, could you please add them?
Also, could you please track 2 issues in the repo:

a) For we address the TODO: https://github.com/kubernetes-sigs/controller-tools/pull/536/files#r1022245967
b) To check further if there is anything else we could replace here in a follow-up to use the upstream implementations instead of re-implement them?